docs/use-cases/deploy-code.md

```diff
@@ -1,8 +1,73 @@
 ---
-title: "Manage and deploy code versions for an existing machine"
+title: "Manage and deploy code versions"
 linkTitle: "Manage and deploy code versions"
-weight: 40
 type: "docs"
-layout: "empty"
-canonical: "registry/"
+weight: 30
+description: "Deploy and manage module versions as public or private resources with the Viam CLI."
 ---
+
+While Viam provides built-in support for a variety of components and services, you can use {{< glossary_tooltip term_id="module" text="modules" >}} to extend support for any hardware components or software services,across both industrial and consumer domains, whether proprietary or not.
+
+Modules allow you to write and deploy custom code to your machine or fleet of machines, with robust control over how changes to your module's code are distributed to deployed machines.
+
+You can choose to develop your module locally, and deploy to your machine directly using a local module installation, or you can upload your module to the Viam registry, making it easy to deploy to all of your machines.
+Further, you can determine the accessibility of your module when you upload it - either keeping it private for exclusive access within your organization or making it public for availability to all Viam users.
+
+When you deploy a module, whether its one you've written yourself or added from the registry, you maintain complete control over how that module's code is deployed to your machine when new updates to the module are available.
+
+<table>
+  <tr>
+    <th>{{<imgproc src="/ml/collect.svg" class="fill alignleft" style="max-width: 150px" alt="ml collect icon">}}
+      <b>1. Search modules</b>
+      <br><br>
+      <p>Once you <a href="/fleet/machines/#add-a-new-robot">have created a machine in the Viam app</a>, <a href="/registry/configure/">search for modules in the Viam registry</a> that fit your machine's requirements, and then <a href="/registry/configure/#add-a-modular-resource-from-the-viam-registry">add a module</a> from your machine's configuration page in the Viam app.</p>
+    </th>
+  </tr>
+  <tr>
+    <th>{{<imgproc src="/ml/configure.svg" class="fill alignleft" style="max-width: 150px" declaredimensions=true alt="ml configure icon">}}
+      <b>2. Create your own module</b>
+      <br><br>
+      <p>Or, you can <a href="https://docs.viam.com/registry/create/">create your own module</a> to add support for new hardware, or to extend an existing software service.</p>
+    </th>
+  </tr>
+  <tr>
+    <th>{{<imgproc src="/ml/deploy.svg" class="fill alignleft" style="max-width: 150px" declaredimensions=true alt="ml deploy icon">}}
+      <b>3. Deploy your module</b>
+      <br><br>
+       <p>Once you have created your new module, you can deploy it to your machine in one of two ways:</p><br><br>
```

Comment:
```suggestion
       <p>Once you have created your new module, you can deploy it to your machine in one of two ways:</p><br><br><br>
```

Think we need one more for larger screens. We'll be fixing this in the near future!